cosmwasm/contracts/crosschain-swaps/src/execute.rs

```diff
@@ -1,24 +1,92 @@
 use cosmwasm_std::{coins, to_binary, wasm_execute, BankMsg, Env, Timestamp};
 use cosmwasm_std::{Addr, Coin, DepsMut, Response, SubMsg, SubMsgResponse, SubMsgResult};
-use registry::Registry;
+use registry::msg::{Callback, SerializableJson};
+use registry::{Registry, RegistryError};
 use swaprouter::msg::ExecuteMsg as SwapRouterExecute;
 
 use crate::checks::{check_is_contract_governor, ensure_key_missing, validate_receiver};
 use crate::consts::{MsgReplyID, CALLBACK_KEY};
-use crate::msg::{CrosschainSwapResponse, FailedDeliveryAction, SerializableJson};
+use crate::msg::{CrosschainSwapResponse, FailedDeliveryAction};
 use registry::proto::MsgTransferResponse;
 
-use crate::state;
 use crate::state::{
     Config, ForwardMsgReplyState, ForwardTo, SwapMsgReplyState, CONFIG, FORWARD_REPLY_STATE,
     INFLIGHT_PACKETS, RECOVERY_STATES, SWAP_REPLY_STATE,
 };
 use crate::utils::{build_memo, parse_swaprouter_reply};
 use crate::ContractError;
+use crate::{state, ExecuteMsg};
+
+/// This function takes any token. If it's already something we can work with
+/// (either native to osmosis or native to a chain connected to osmosis via a
+/// valid channel), it will just proceed to swap and forward. If it's not, then
+/// it will send an IBC message to unwrap it first and provide a callback to
+/// ensure the right swap_and_forward gets called after the unwrap succeeds
+pub fn unwrap_or_swap_and_forward(
+    deps: DepsMut,
+    block_time: Timestamp,
+    contract_addr: Addr,
+    swap_coin: Coin,
+    output_denom: String,
+    slippage: swaprouter::Slippage,
+    receiver: &str,
+    next_memo: Option<SerializableJson>,
+    failed_delivery_action: FailedDeliveryAction,
+) -> Result<Response, ContractError> {
+    deps.api
+        .debug(&format!("executing unwrap or swap and forward"));
```

Comment:
Purely informational question, do these debugs get into the final cosmwasm binary? Or are they only in local development

Reply:
good question 🤔 . They're not printed, but not sure if they get removed. 

They must be in the binary, since the one we sue for testing is the same as the release one. My guess is that the call still exists (it's part of the Api trait) but they're a no-op. Still the string generation probably happens

Reply:
In this case, hopefully the format macro gets no-op'd, and the string content just appears in the static portion of the binary

Reply:
I'd guess the format is still executed unless the compiler does something smart there. But we can also remove those before releasing

Reply:
meh, doubt this is going to be make or break for anything. Was mostly just curious